From ad0a04125051a4638a31e6e0c2d1dc711676e1ef Mon Sep 17 00:00:00 2001
From: Mike Massonnet <mmassonnet@xfce.org>
Date: Wed, 8 Sep 2010 22:01:50 +0200
Subject: [PATCH] Launch xfce4-session with ck-launch-session and dbus-launch.

Makes shutdown/reboot and mount/unmount work for gvfs when login through
a tty.
---
 scripts/xinitrc.in.in |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/scripts/xinitrc.in.in b/scripts/xinitrc.in.in
index 2bd43ec..241d00f 100755
--- a/scripts/xinitrc.in.in
+++ b/scripts/xinitrc.in.in
@@ -196,7 +196,11 @@ fi
 
 # Run xfce4-session if installed
 if which xfce4-session >/dev/null 2>&1; then
-    xfce4-session
+    if test -z "$XDG_SESSION_COOKIE" -a x"`which ck-launch-session 2>/dev/null`" != x""; then
+        ck-launch-session dbus-launch --sh-syntax --exit-with-session xfce4-session
+    else
+        xfce4-session
+    fi
 
     if test "$ssh_agent_kill_cmd"; then
         echo "running '$ssh_agent_kill_cmd'"
-- 
1.7.2.3

